atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -348,6 +349,11 @@ private NonEvictionLoggingEvictionPolicy(EvictionPolicy<T> delegate) {
 
         @Override
         public boolean evict(EvictionConfig config, PooledObject<T> underTest, int idleCount) {
+            if (log.isDebugEnabled()) {
+                log.debug("Beginning an eviction attempt from the Cassandra client pool",
+                        SafeArg.of("underTestState", underTest.getState()),
+                        SafeArg.of("idleState", underTest.getIdleTimeMillis()));
+            }
             boolean delegateResult = delegate.evict(config, underTest, idleCount);
```

Comment:
Do we want to wrap with `TimedRunner`?

Reply:
We could, but I think it's probably not needed. The delegate impl looks like this:

```
    @Override
    public boolean evict(final EvictionConfig config, final PooledObject<T> underTest,
            final int idleCount) {

        if ((config.getIdleSoftEvictTime() < underTest.getIdleTimeMillis() &&
                config.getMinIdle() < idleCount) ||
                config.getIdleEvictTime() < underTest.getIdleTimeMillis()) {
            return true;
        }
        return false;
    }
```

so this should be a bunch of memory reads. Actually this starting log block isn't probably necessary in fact.